lib/signalfx.js

```diff
@@ -0,0 +1,334 @@
+'use strict';
+// Copyright (C) 2015 SignalFx, Inc. All rights reserved.
+
+(function () {
+
+  var request = require('request'); // Use for create post request
+  var async = require('async'); // Use fo make post request in separate thread
+  var winston = require('winston'); // Logging library
+
+// Application version and name for User-Agent
+  var name = 'signalfx-nodejs-client';
+  var version = '0.1.0';
+
+// Default Parameters
+  var DEFAULT_INGEST_ENDPOINT = 'https://ingest.signalfx.com';
+  var DEFAULT_API_ENDPOINT = 'https://api.signalfx.com';
+  var DEFAULT_BATCH_SIZE = 300;// Will wait for this many requests before posting
+  var DEFAULT_TIMEOUT = 1500; // Default timeout is 1s
+
+// Global Parameters
+  var PROTOBUF_HEADER_CONTENT_TYPE = 'application/x-protobuf';
+  var JSON_HEADER_CONTENT_TYPE = 'application/json';
+
+
+// ============================================================================================
+//                            Check if we can use Protobuf
+// ============================================================================================
+  var useProtocolBuffers = true;
+  var protocolBuffers;
+  try {
+    protocolBuffers = require('./signal_fx_protocol_buffers_pb2').com.signalfx.metrics.protobuf;
+  } catch (error) {
+    winston.warn('Protocol Buffers not installed properly. Switching to Json.', error);
+    useProtocolBuffers = false;
+  }
+
+// ============================================================================================
+//                            BaseSignalFx Signal Fx class
+// ============================================================================================
+  function BaseSignalFx(apiToken, ingestEndpoint,
```

Comment:
What's the distinction in role between BaseSignalFx and SignalFxClient? This type doesn't seem to provide any meaningful abstraction which calls for separating it out from SignalFxClient. Did you have other use cases for this base class in mind?